### PR TITLE
Change XML operations configuration to support turning off all operations

### PR DESCRIPTION
**Pull request by sawmurai**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/api-platform/api-platform/issues/252
| License       | MIT
| Doc PR        | https://github.com/api-platform/docs/pull/185

*Please update this template with something that matches your PR*

### CODE REVIEW

File: src/Metadata/Extractor/XmlExtractor.php [hunk 1]
```diff
@@ -43,11 +43,23 @@ protected function extractPath(string $path)
                 'shortName' => $this->phpize($resource, 'shortName', 'string'),
                 'description' => $this->phpize($resource, 'description', 'string'),
                 'iri' => $this->phpize($resource, 'iri', 'string'),
-                'itemOperations' => $this->getAttributes($resource, 'itemOperation') ?: null,
```
Comment (meyerbaptiste) [inline src/Metadata/Extractor/XmlExtractor.php]: IMO you should create a dedicated method like this `XmlExtractor::getOperations($resource, $operationType)`.

File: src/Metadata/Extractor/XmlExtractor.php [hunk 2]
```diff
@@ -43,15 +43,33 @@ protected function extractPath(string $path)
...
+     */
+    private function getOperations(\SimpleXMLElement $resource, string $operationType)
+    {
+        $operationsParent = $operationType . 's';
+
+        if (!isset($resource->$operationsParent)) {
+            return null;
+        }
+
+        return $this->getAttributes($resource->$operationsParent, $operationType);;
```
Comment (soyuka) [inline src/Metadata/Extractor/XmlExtractor.php]: ;;?

File: src/Metadata/Extractor/XmlExtractor.php [hunk 3]
```diff
@@ -43,15 +43,33 @@ protected function extractPath(string $path)
...
     /**
+     * Return the array containing configured operations. Returns NULL if there is no operation configuration.
+     *
+     * @param \SimpleXMLElement $resource
+     * @param string            $operationType
+     * @return array|null
+     */
+    private function getOperations(\SimpleXMLElement $resource, string $operationType)
+    {
+        $operationsParent = $operationType . 's';
```
Comment (soyuka) [inline src/Metadata/Extractor/XmlExtractor.php]: no spaces surrounding `.`

**Comment by meyerbaptiste** [comment]
Currently it's a huge BC break but what do you think about keeping the old configuration, deprecating it and basing his PR on the `2.0` branch?

**Comment by sawmurai** [comment]
Sure, no problem. So I would only edit this PR and base it on the 2.0 branch, right? Or anything additional regarding the deprecation? 

Personally I circumvented the problem by providing the config via Yaml.

**Comment by sawmurai** [comment]
Changed back to master ... if you need a new PR against 2.0 please let me know. :)

**Comment by soyuka** [comment]
> Currently it's a huge BC break but what do you think about keeping the old configuration, deprecating it and basing his PR on the 2.0 branch?

ref https://github.com/api-platform/api-platform/issues/252#issuecomment-286700475

This is indeed a huge BC break. Keeping both behaviors is messy IMO but might be best if we want to see this fixed ASAP.  

@sawmurai You should not merge back target branches in feature branches :p. Use rebase instead [see here](https://github.com/api-platform/core/pull/975#issuecomment-285621784) to revert the last merge.

If you want to target 2.0 so that it can be merged sooner, you'd have to:

1. undo the changes in old tests (usually if tests changes, it's a BC break)
2. make sure that the tests still pass with the fix (add @trigger_error for deprecation messages [see here](https://github.com/api-platform/core/pull/904/files#diff-c38359c69731b3bdcea477d3e35c1f97R63)).
3. make the `nocollectionoperations.xml` test pass
4. rebase this branch on 2.0 (git rebase 2.0)

**Comment by sawmurai** [comment]
Sorry guys I completely messed this one up ... here is the new one for 2.0:

https://github.com/api-platform/core/pull/993

**Pull request closed without merging**